storage/filesystem/dotgit/writers_test.go

```diff
@@ -161,3 +162,68 @@ func TestPackWriterUnusedNotify(t *testing.T) {
 
 	assert.NoError(t, w.Close())
 }
+
+func TestPackWriterPermissions(t *testing.T) {
+	t.Parallel()
+
+	if runtime.GOOS == "windows" {
+		t.Skip("not applicable to windows")
+	}
+
+	f := fixtures.Basic().One()
+
+	fs := osfs.New(t.TempDir(), osfs.WithBoundOS())
+	dot := New(fs)
+	require.NoError(t, dot.Initialize())
+
+	w, err := dot.NewObjectPack()
+	require.NoError(t, err)
+
+	_, err = io.Copy(w, f.Packfile())
+	require.NoError(t, err)
+
+	require.NoError(t, w.Close())
+
+	pfPath := fmt.Sprintf("objects/pack/pack-%s.pack", f.PackfileHash)
+	idxPath := fmt.Sprintf("objects/pack/pack-%s.idx", f.PackfileHash)
+	revPath := fmt.Sprintf("objects/pack/pack-%s.rev", f.PackfileHash)
+
+	stat, err := fs.Stat(pfPath)
+	require.NoError(t, err)
+	assert.Equal(t, os.FileMode(0o644), stat.Mode().Perm())
+
+	stat, err = fs.Stat(idxPath)
+	require.NoError(t, err)
+	assert.Equal(t, os.FileMode(0o644), stat.Mode().Perm())
```

Comment:
The `0o644` permission does not seem to align with the upstream [code](https://github.com/git/git/blob/d8af7cadaa79d5837d73ec949e10b57dedb43e9b/builtin/index-pack.c#L1605) nor my observation of repos I cloned, which aligns with `0o444`:

<img width="874" height="82" alt="Image" src="https://github.com/user-attachments/assets/e634078c-0c3c-45be-8fb9-b7b0188f4ebc" />

Reply:
You are right, it should be 444.